presto-main/src/test/java/com/facebook/presto/sql/query/TestSubqueries.java

```diff
@@ -118,15 +118,35 @@ public void testCorrelatedSubqueriesWithLimit()
                 "select EXISTS(select 1 from (values 1, 1, 3) t(a) where t.a=t2.b limit 1) from (values 1, 2) t2(b)",
                 "VALUES true, false",
                 false);
-        // TransformCorrelatedScalarAggregationToJoin does not fire since limit is above aggregation node
+        assertions.assertQuery(
+                "select (select count(*) from (values 1, 1, 3) t(a) where t.a=t2.b group by a limit 1) from (values 1.0) t2(b)",
+                "VALUES BIGINT '2'");
         assertions.assertFails(
-                "select (select count(*) from (values 1, 1, 3) t(a) where t.a=t2.b limit 1) from (values 1) t2(b)",
+                "SELECT (SELECT count(*) FROM (VALUES 1, 1, 3) t(a) WHERE t.a=t2.b LIMIT 1) FROM (VALUES 1) t2(b)",
                 UNSUPPORTED_CORRELATED_SUBQUERY_ERROR_MSG);
         assertExistsRewrittenToAggregationBelowJoin(
                 "SELECT EXISTS(SELECT 1 FROM (values ('x', 1)) u(x, cid) WHERE x = 'x' AND t.cid = cid LIMIT 1) " +
                         "FROM (values 1) t(cid)",
                 "VALUES true",
                 false);
+        // Test decorrelated subexpression with ArithmeticBinaryExpression
+        assertions.assertQuery(
+                "select (select count(*) from (values 1, 1, 3) t(a) where t.a = t2.a + t2.b group by a limit 1) from (values (1.0, 1.0),(1.0,2.0)) t2(a,b)",
+                "VALUES null, BIGINT '1'");
+        // Test decorrelated subexpression with ArithmeticUnaryExpression
+        assertions.assertQuery(
+                "select (select count(*) from (values 1, 2, 3,4,5) t(a) where t.a = t2.a + (-t2.b)+1 group by a limit 1) from (values (1.0, 1.0),(1.0, 2.0),(1.0,2.0)) t2(a,b)",
+                "VALUES null, null, BIGINT '1'");
+        // Test decorrelated subexpression with NotExpression
+        assertions.assertQuery(
```

Comment:
I think this needs an order by to be deterministic.  if it's `ORDER BY a DESC`, then the answer is `1` (the row where a=3), but if it's `ORDER BY a ASC`, then the answer is `2`  (the two rows where a=1).  It would better though to leave out the group by + limit.

Reply:
Are you suggesting to do `ORDER BY` inside the subquery? The ordering is unnecessary because the query returns a final count. sure I will leave out the groupby + limit clause

Reply:
The ordering is necessary because the query returns a count grouped by a, and there are 2 values of a.  Without the ordering, it's not deterministic which count will be returned.    By the way, I think I meant just leave out the limit, though leaving out both group by and limit is okay too.  Also, the outermost select is not necessary in this query.

Reply:
I see, that's a good point, thank you, I will fix the ut now with `order by` clause to make it clear

Reply:
The order by isn't needed anymore since you removed the limit.  It was only needed so it could be deterministic which one value would be returned.

Reply:
apology for the confusion. I took out the `order by` first and then added it back but left out limit. Anyway, I'm going to add `order by asc` and `order by desc` both cases